src/Export/ExportCsv.php

```diff
@@ -46,21 +45,20 @@ private function getExportCallback(
 			array $data,
 			DataGrid $grid
 		) use ($name, $outputEncoding, $delimiter, $includeBom): void {
-			$columns = $this->getColumns() ?: $this->grid->getColumns();
+			$columns = $this->getColumns();
+			if ($columns === []) {
+				$columns = $this->grid->getColumns();
+			}
```

Comment:
Looks like getColumns() and grid->getColumns() returns the same, shouldn't getColumns() call and cache grid->getColumns() internally? cc @paveljanda

Reply:
I don't think so. There could be custom columns in ExportCsv set.